### ISSUE TITLE
Feature Request: Grid-List Primary Action, Action Positions

### ISSUE THREAD

**Issue by MyNameIsCosmo** [material-components/material-components-web#2317]
According to the [material.io grid-list guidlines](https://material.io/guidelines/components/grid-lists.html), the grid-list tiles have a primary action as well as secondary actions positioned within each corner of the tile.

These features are especially useful within a gallery selection.

I will be looking into providing an engineering outline document and a pull request sometime this weekend unless otherwise discussed.

**Comment by kfranqueiro** [comment]
Hey, thanks for providing the initial heads-up on your interest here!

We're looking at redesigning Grid List fairly soon, into a component with a more focused use case (see #2314 and #2315 for its first tasks), and it's likely that the alternate action corners won't be a thing (or at least not a priority) in the new component. It's also likely that we'll prioritize actionable iconography later in general, due to potential intricacies with it overlapping the overall item's action area.

If you are interested in investigating how complex this would be once actionable iconography lands in the new component, we can pick up this conversation then. Thanks!

**Comment by kfranqueiro** [comment]
Hey, sorry for the delay in reply. I just attended another meeting between engineering and design today, and have a document I need to clean up to reflect a way I realized I can simplify what I was planning RE mixins for Image List. I'll get a public link to it once that's taken care of.

**Comment by MyNameIsCosmo** [comment]
No need to apologize!
Keep me updated, I'd love to help out where I can.

Cheers!

**Comment by kfranqueiro** [comment]
I had some time to clean up the document.

* [Design Document](https://docs.google.com/document/d/1Ns5QhV2PckT-fRsgYQGOY0F5fHGvSwaowxCUcSPK8QI/edit?usp=sharing)
* [Prototype](https://codepen.io/kfranqueiro/pen/qxqxdz)

#2367 is the first PR, and I have another branch ready to add Masonry Image List once that's merged. The prototype contains partial work towards some of the later tasks, but there's more fleshing out to do there.

Pertinent to this issue, the text label and actionable iconography have been simplified for Image List compared to the existing Grid List, and there's no longer any design provision for putting actions in various/multiple corners.

However, if you can present a use case and would be interested in working on it once we've got enough of Image List committed, then perhaps we can run it by our designer as well. (I mentioned that this issue existed and our designer seemed open to considering it.)

Thanks!
